### ISSUE TITLE
Mouse wheel change desktop also in main window

### ISSUE THREAD

**Issue by alejandro-perez** [gmc-holle/xfdashboard#41]
In GS when you use the wheel on the main overview window, the virtual desktop changes (in the same way it does when you use it on the right panel. Is it possible to replicate this behaviour?

**Comment by gmc-holle** [comment]
Now key and mouse binding configuration is implemented at their basics this issue should be resolved for next stable release.

**Comment by gmc-holle** [comment]
Sadly key and mouse binding configuration is just a key-binding only :( So this feature will not be configurable via bindings.

**Comment by gmc-holle** [comment]
With commit b753c9943968a25d724a8e6b178f8c07acd01f8d this is now configurable via a xfconf property called "/components/windows-view/scroll-event-changes-workspace". This property is a boolean and must be created if it does not exists. Set this property to TRUE to allow mouse-wheel events to change active workspace like it does in workspace selector.

Closing issue.

**Comment by alejandro-perez** [comment]
I does not work for me anymore. I'm not sure whether this is caused by this commit or not, since I haven't used xfdashboard for a while. 
What it happens is the following: I start xfdashboard and nothing appears. However, it is running, and I cannot click on the windows of my desktop. I have to press ESC to close xfdashboard and then I can continue. Would you be interested in seeing the log output?

Regards

**Comment by gmc-holle** [comment]
Bad timing. I just release a new development version ;) But a bug is a bug so let's so go on.

When xfdashboard starts it does not display anything? The screen is completely blank? Are you using a customized theme? Because with this new version there was a change in theming. Please remove or rename any customized themes in ${HOME}/.local/share/themes and ${HOME}/.themes which affect xfdashboard and try again. Also please revert the default value "xfdashboard" in xfconf property "/theme" in group (or channel) xfdashboard. The easiest way is to use 'xfce4-settings-editor', select "xfdashboard" left in the channels list, then mark 'theme' and click on "Reset".

If you don't use any customized theme then please tell me what distribution you are using and which version of Clutter, Glib and Gtk+ is installed.

And thanks for trying even if you do not use xfdashboard :)

**Comment by alejandro-perez** [comment]
Hey, you know your stuff :). Indeed it was the current theme. I customized it because I like the side panel to be always visible. After having removed it it has worked.

There is something different I think. Nor better or worse, but different. When the "overview" is being shown, the background is the actual desktop, with all the windows opened. Is that intended? (see attachment)
![screencap](https://cloud.githubusercontent.com/assets/2082536/5691577/8b303f80-98ce-11e4-8965-6724ef517492.png)

BTW, great work. I was not using it mainly because of the mouse binding. I will probably use hereafter :).

**Comment by gmc-holle** [comment]
Yeah, I guessed right :D

You can re-use your old customized theme but you have to change one word. Just locate the file "xfdashboard.xml" in your theme's folder and replace "XfdashboardStage" with "XfdashboardStageInterface". This word should occur only once. Then it should work again. Sorry for this inconvinience.

No, it is not intended show the actual desktop with all windows opened. It is intended that only the desktop with your wallpaper is shown. It seems that xfdashboard find the "wrong" root window which should be the desktop with the wallpaper on it. What window manager are you using? Have you set any wallpaper?

**Comment by alejandro-perez** [comment]
First, I did what you said but nothing changed. What should I put in the /theme property? How should I call the folder in .local/share/themes/xfdashboard/? If I use xfdashboard then it works, but not my theme. If I set them to "alex" it says that the theme does not exist.

Second. The behaviour I have told you only happens in daemon mode. Besides, it shows every window openeed behind, or the desktop if there is none. See snapshot of more windows.
![snapshot](https://cloud.githubusercontent.com/assets/2082536/5691641/4d4fa242-98d2-11e4-9641-e25b4233329f.png)

**Comment by gmc-holle** [comment]
It's easier for me to make checklist. So please apologize the maybe aggresive list.

If you name your theme "alex" then ...
1.) .. there should be folder called "alex" either at ${HOME}/.local/share/themes/alex or at ${HOME}/.themes/alex (what you prefer)
2.) In this folder there should be another folder called "xfdashboard-1.0" containing all the files needed for theme, i.e. xfdashboard.theme, the layout file (usually xfdashboard.xml), the css file (usually at least xfdashboard.css) and maybe effect xml file, images etc.
3.) Remember to rename XfdashboardStage to XfdashboardStageInterface in the layout file (usually xfdashboard.xml)
4.) The property "theme" should be set to "alex"
5.) That's all

If you running xfdashboard in daemon mode then either quit it with "xfdashboard -q" at console and restart it or change the property theme to "xfdashboard" and then back to "alex" to re-apply theme.

I try to reproduce the second issue with the desktop and all windows in the background. If I can reproduce it, I open a new issue. Because this is a completely different issue which is not related to theming.

**Comment by alejandro-perez** [comment]
Actually, I prefer this kind of exhaustive lists, thanks.
The theming part worked, but then the background windows issues becomes even more annoying, since background windows are not grayed out, and they appear even without daemon mode.

![snapshot](https://cloud.githubusercontent.com/assets/2082536/5691749/897c1b16-98d6-11e4-8e39-7699bb3fb1d5.png)

Here is the theme:
https://www.dropbox.com/s/e51uz5rvaelr8hj/xfdashboard-theme-alex.tgz?dl=0

Regards

**Comment by gmc-holle** [comment]
Stupid me. Of course you have to replace XfdashboardStage with XfdashboardStageInterface in your CSS files. Sorry :( Remember to change the property theme to "xfdashboard" and then back to "alex" to re-apply theme or restarting xfdashboard.

**Comment by gmc-holle** [comment]
Could you solve this issue?

**Comment by alejandro-perez** [comment]
No, it didn't work the last time I tried (a few weeks ago). Nevertheless, I have temporary switched to KDE5, so I won't be able to test it (unless I decided to switch back, which I don't discard).

If none else is complaining, you can close the issue.

Thanks for all the effort and the good work!

**Comment by gmc-holle** [comment]
Thanks for your feedback. Then I will close this issue.

Thanks for trying xfdashboard :)

**Comment by alejandro-perez** [comment]
Hi,

I switched back to XFCE and wanted to test this. Still  having the same problem with the background windows. It shouldn't be just me as the snapthost of the default theme also shows the same problem :). Windows are shown in the background (greyed out), but it should show  the wallpaper instead.

Do you want me to open a new issue (since the title doesn't match)?

Regards

**Comment by gmc-holle** [comment]
I agree. It should be the best idea to open a new issue for this problem. Could you also please check that no folder called "xfdashboard" (the name of the default) exists at $HOME/.themes and $HOME/.local/share/themes?

Thanks

**Comment by alejandro-perez** [comment]
Done. Anyway, those folders are empty. Note that the snapshot from the theme shows the issue.
